camstack/cams/cred1.py

```diff
@@ -474,55 +473,66 @@ class Apapane(CRED1):
         'CTYPE1': ('RA--TAN   ', 'Pixel coordinate system', '%-10s', 'CTYP1'),
         'CTYPE2': ('DEC--TAN  ', 'Pixel coordinate system', '%-10s', 'CTYP2'),
+
+        'C2ELT1': (4.5e-6, 'X Scale projected on detector (#/pix)', '%20.8f', 'C2EL1'),
+        'C2ELT2': (4.5e-6, 'Y Scale projected on detector (#/pix)', '%20.8f', 'C2EL2'),
         'C2YPE1': ('RA--TAN   ', 'Pixel coordinate system', '%-10s', 'C2YP1'),
         'C2YPE2': ('DEC--TAN  ', 'Pixel coordinate system', '%-10s', 'C2YP2'),
-        'CUNIT1': ('DEGREE    ', 'Units used in both CRVAL1 and CDELT1', '%-10s', 'CUNI1'),
-        'CUNIT2': ('DEGREE    ', 'Units used in both CRVAL2 and CDELT2', '%-10s', 'CUNI2'),
         'C2NIT1': ('DEGREE    ', 'Units used in both C2VAL1 and C2ELT1', '%-10s', 'C2NI1'),
         'C2NIT2': ('DEGREE    ', 'Units used in both C2VAL2 and C2ELT2', '%-10s', 'C2NI2'),
 
         # Those will change with cropmode
         'CRPIX1': ( 40., 'Reference pixel in X (pixel)', '%20.1f', 'CRPX1'),
         'CRPIX2': ( 80., 'Reference pixel in Y (pixel)', '%20.1f', 'CRPX2'),
-        'C2PIX1': (120., 'Reference pixel in X (pixel)', '%20.1f', 'C2PX1'),
-        'C2PIX2': ( 80., 'Reference pixel in Y (pixel)', '%20.1f', 'C2PX2'),
-
         'CD1_1': (4.5e-6, 'Pixel coordinate translation matrix', '%20.8f', 'CD1_1'),
         'CD1_2': (    0., 'Pixel coordinate translation matrix', '%20.8f', 'CD1_2'),
         'CD2_1': (    0., 'Pixel coordinate translation matrix', '%20.8f', 'CD2_1'),
         'CD2_2': (4.5e-6, 'Pixel coordinate translation matrix', '%20.8f', 'CD2_2'),
-    }
+
+        'C2PIX1': (120., 'Reference pixel in X (pixel)', '%20.1f', 'C2PX1'),
+        'C2PIX2': ( 80., 'Reference pixel in Y (pixel)', '%20.1f', 'C2PX2'),
+        'C21_1': (4.5e-6, 'Pixel coordinate translation matrix', '%20.8f', 'CD1_1'),
+        'C21_2': (    0., 'Pixel coordinate translation matrix', '%20.8f', 'CD1_2'),
+        'C22_1': (    0., 'Pixel coordinate translation matrix', '%20.8f', 'CD2_1'),
+        'C22_2': (4.5e-6, 'Pixel coordinate translation matrix', '%20.8f', 'CD2_2'),
+    '''
     # yapf: enable
     KEYWORDS.update(CRED1.KEYWORDS)
 
     REDIS_PUSH_ENABLED = True
-    REDIS_PREFIX = 'x_B'  # LOWERCASE x to not get mixed with the SCExAO keys
+    REDIS_PREFIX = 'x_A'  # LOWERCASE x to not get mixed with the SCExAO keys
 
     def _constructor_finalize(self) -> None:
         self.send_command('set imagetags on')
         self.send_command('set rawimages on')
 
     def _fill_keywords(self) -> None:
-        CRED1._fill_keywords(self)
-
+        # First, determine how many WCS we need.
         doing_pdi = True
-
         if self.HAS_REDIS:
             try:
                 self._set_formatted_keyword('FILTER01',
                                             self.RDB.hget('X_IRCFLT', 'value'))
+                # FIXME
                 doing_pdi = self.RDB.hget('OBS-MOD', 'value') == 'IMAG-PDI'
```

Comment:
Should any single camera be setting the obs-mod in the redis db? That seems prone to clashes between instruments